dashboard/test/lib/pd/payment/payment_calculator_test.rb

```diff
@@ -33,6 +42,62 @@ class PaymentCalculatorTest < ActiveSupport::TestCase
       assert_equal 0, PaymentCalculator.instance.calculate(insufficient_puzzles)
     end
 
+    test 'Error raised if there is no payment term for workshop' do
+      error = assert_raises(RuntimeError) do
+        PaymentCalculator.instance.calculate(@csp_workshop)
+      end
+
+      assert_equal "No payment terms were found for workshop #{@csp_workshop.id}", error.message
+    end
+
+    test 'Calculate CSP Workshop with per attendee payment' do
+      create(:pd_payment_term, regional_partner: @regional_partner, properties: {per_attendee_payment: 10})
+
+      assert_equal 200, PaymentCalculator.instance.calculate(@csp_workshop)
+    end
+
+    test 'Calculate CSP Workshop with fixed payment' do
+      create(:pd_payment_term, regional_partner: @regional_partner, properties: {fixed_payment: 42})
+
+      assert_equal 42, PaymentCalculator.instance.calculate(@csp_workshop)
+    end
+
+    test 'Calculate CSP Workshop with per attendee payment and fixed payment' do
+      create(:pd_payment_term, regional_partner: @regional_partner, properties: {fixed_payment: 42, per_attendee_payment: 10})
+
+      assert_equal 242, PaymentCalculator.instance.calculate(@csp_workshop)
+    end
+
+    test 'Calculate CSP Workshop with excessive teachers' do
+      create(:pd_payment_term, regional_partner: @regional_partner, properties: {per_attendee_payment: 10, maximum_attendees_for_payment: 19})
+
+      assert_equal 190, PaymentCalculator.instance.calculate(@csp_workshop)
+    end
+
+    test 'Calculate CSP Workshop with insufficient teachers' do
+      payment_term = create(:pd_payment_term, regional_partner: @regional_partner, properties: {per_attendee_payment: 10, minimum_attendees_for_payment: 30})
+
+      assert_equal 0, PaymentCalculator.instance.calculate(@csp_workshop)
+
+      payment_term.destroy
+
+      create(:pd_payment_term, regional_partner: @regional_partner, properties: {per_attendee_payment: 10, minimum_attendees_for_payment: 30, fixed_payment: 42})
+
+      assert_equal 42, PaymentCalculator.instance.calculate(@csp_workshop)
+    end
+
+    test 'Calculate CSP Workshop with sufficient attendees and facilitator payment' do
+      create(:pd_payment_term, regional_partner: @regional_partner, properties: {per_attendee_payment: 10, facilitator_payment: 50})
+
+      assert_equal 300, PaymentCalculator.instance.calculate(@csp_workshop)
+    end
+
+    test 'Calculate CSP Workshop with insufficient attendees and facilitator payment' do
+      create(:pd_payment_term, regional_partner: @regional_partner, properties: {fixed_payment: 42, minimum_attendees_for_payment: 30, facilitator_payment: 50})
+
+      assert_equal 42, PaymentCalculator.instance.calculate(@csp_workshop)
+    end
+
```

Comment:
Oh right. I was thinking of the old model. Still it can't hurt. 

Are we no longer concerned with minimum attendance days? If a teacher attends 1 session in a 5-day workshop, are they meant to count as attended?